src/main/java/com/jcabi/w3c/DefaultValidationResponse.java

```diff
@@ -161,11 +169,13 @@ public void addWarning(final Defect warning) {
      * @return The text
      */
     private String asText(final Set<Defect> defects) {
-        final StringBuilder text = new StringBuilder(0);
-        for (final Defect defect : defects) {
-            text.append("  ").append(defect.toString()).append('\n');
+        synchronized (defects) {
```

Comment:
@piddubnyi we don't need the synchronized block to read from CopyOnWriteArraySet